design/aws-ecs-service-discovery-integration.md

```diff
@@ -0,0 +1,235 @@
+# AWS ECS - Support for AWS Cloud Map (Service Discovery) Integration
+
+To address issue [#1554](https://github.com/awslabs/aws-cdk/issues/1554), the
+ECS construct library should provide a way to set
+[`serviceRegistries`](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html#ECS-CreateService-request-serviceRegistries)
+on their ECS service at the L1 level.
+
+
+## General approach
+Rather than having the customer instantiate Cloud Map constructs directly and
+thus be forced to learn the API of a separate construct library, we should
+allow the customer to pass in the domain name and other configuration minimally
+needed to instantiate a Cloud Map namespace within an ECS cluster and create a
+Cloud Map service for each ECS service.
+
+The current proposal is to add a method on the ECS Cluster construct,
+`addNamespace` that would take a set of properties that include the namespace
+type (Public DNS, Private DNS or Http) and namespace name (domain name).
+
+While it is possible to use more than one namespace for services in an ECS
+cluster, realistically, we would expect ECS customers to only have one
+namespace per given cluster. A Cloud Map service within that namespace would
+then be created for each ECS service using service discovery in the cluster,
+and would be discoverable by service name within that namespace, e.g.
+frontend.mydomain.com, backend.mydomain.com, etc.
+
+ECS will automatically register service discovery instances that are accessible
+by IP address (IPv4 only) on the createService API call.
+
+// FIXME public namespace needs to be imported?
```

Comment:
?

Reply:
I'm not quite sure yet what the correct workflow is if the customer chooses to bring their own public DNS namespace here, since this is not something that would be owned by Route53.